components/tree/nz-tree.component.ts

```diff
@@ -40,6 +40,7 @@ export class NzTreeComponent implements OnInit {
   @Input() nzDefaultExpandAll: boolean = false;
   @Input() nzDefaultExpandedKeys: string[] = [];
   @Input() nzDefaultSelectedKeys: string[] = [];
+  @Input() nzTreeSelectTemplate: TemplateRef<{}>;
   @Input() nzBeforeDrop: (confirm: NzFormatBeforeDropEvent) => Observable<boolean>;
```

Comment:
No need to add this input, you can use `nzTreeTemplate` directly.

Reply:
`nzTreeTemplate` can't deliver from `tree-select` to `tree` without using input : 

**nz-tree-select.component**
```html
<nz-tree #treeRef ...>
  <ng-container [ngTemplateOutlet]="nzTreeSelectTemplate"></ng-container>
</nz-tree>
```
I tried but `tree component` can't get its `@ContentChild`, do you mean this ?